gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestPlanMapperTest.java

```diff
@@ -120,212 +198,495 @@ public void When_MappingAuthoriseResourceWithAllOptionals_Expect_AllFieldsToBeMa
         var statusReason = (CodeableConcept) statusReasonExt.get(0).getValue();
 
-        assertThat(statusReason.getText()).isEqualTo("Patient no longer requires these");
+        assertAll(
+            () -> assertThat(statusExt).hasSize(1),
+            () -> assertThat(statusReasonExt).hasSize(1),
+            () -> assertThat(statusReason.getText()).isEqualTo("Patient no longer requires these")
+        );
     }
 
     @Test
-    public void When_MappingDiscontinue_With_MissingPertinentInformationAndCodeDisplayPresent_Expect_DefaultTextAddedAsReason() {
-        var ehrExtract = unmarshallEhrExtract("ehrExtract7.xml");
-        Optional<RCMRMT030101UKMedicationStatement> medicationStatement = extractMedicationStatement(ehrExtract);
-        Optional<RCMRMT030101UKAuthorise> supplyAuthorise = extractSupplyAuthorise(medicationStatement.orElseThrow());
-
-        var medicationRequest =
-            medicationRequestPlanMapper.mapToPlanMedicationRequest(ehrExtract, medicationStatement.get(), supplyAuthorise.orElseThrow(),
-                PRACTISE_CODE);
+    public void When_MappingDiscontinueWithCodeDisplayAndMissingPertinentInformation_Expect_DefaultTextAddedAsReason() {
+        var medicationStatementXml = """
+            <MedicationStatement xmlns="urn:hl7-org:v3" classCode="SBADM" moodCode="INT">
+                <id root="B4D70A6D-2EE4-41B6-B1FB-F9F0AD84C503"/>
+                <component typeCode="COMP">
+                    <ehrSupplyAuthorise classCode="SPLY" moodCode="INT">
+                        <id root="TEST_ID"/>
+                    </ehrSupplyAuthorise>
+                </component>
+                <component typeCode="COMP">
+                    <ehrSupplyDiscontinue classCode="SPLY" moodCode="RQO">
+                        <id root="D0BF39CA-E656-4322-879F-83EE6E688053"/>
+                        <code code="EMISDRUG_DISCONTINUATION"
+                              codeSystem="2.16.840.1.113883.2.1.6.3"
+                              displayName="Medication Course Ended"
+                        />
+                        <availabilityTime value="20060426"/>
+                        <reversalOf typeCode="REV">
+                            <priorMedicationRef classCode="SBADM" moodCode="ORD">
+                                <id root="TEST_ID"/>
+                            </priorMedicationRef>
+                        </reversalOf>
+                    </ehrSupplyDiscontinue>
+                </component>
+            </MedicationStatement>
+            """;
+        var ehrExtract = unmarshallEhrExtractFromMedicationRequestXml(medicationStatementXml);
+        var medicationStatement = extractMedicationStatement(ehrExtract);
+        var supplyAuthorise = extractSupplyAuthorise(medicationStatement);
 
+        var medicationRequest = medicationRequestPlanMapper.mapToPlanMedicationRequest(
+            ehrExtract,
+            medicationStatement,
+            supplyAuthorise,
+            PRACTISE_CODE
+        );
```

Comment:
Are these 10 lines not repeated a lot in this file, a quick method extract perhaps?